src/dune_pkg/rev_store.mli

```diff
@@ -32,4 +32,8 @@ end
 
 val content_of_files : t -> File.t list -> string list Fiber.t
 val load_or_create : dir:Path.t -> t Fiber.t
-val add_repo : t -> source:string -> Remote.t Fiber.t
+
+(** [add_repo t ~update ~source] idempotently adds a git repo to the rev store.
+    If [update] is [false], it avoids fetching the remote repo, thus if the repo
+    was already previously added this can be done without network I/O *)
+val add_repo : t -> update:bool -> source:string -> Remote.t Fiber.t
```

Comment:
For a good API, I think updating should be a separate function from adding a repository.

One of the issues that we've ran into with an opam API is that it tries to cram every type behavior into a function flag. I'd like to avoid that and keep smaller, simpler functions that can be composed.

Reply:
That's a good point, I removed it from the rev store. I kept it mostly so users of the API don't accidentally forget to run `update`. However, I ended up solving it via types in a nicer way.